segmentstore/server/src/main/java/io/pravega/segmentstore/server/store/ServiceBuilder.java

```diff
@@ -231,13 +235,13 @@ protected SegmentToContainerMapper createSegmentToContainerMapper(ServiceConfig
 
     protected WriterFactory createWriterFactory() {
         WriterConfig writerConfig = this.serviceBuilderConfig.getConfig(WriterConfig::builder);
-        return new StorageWriterFactory(writerConfig, this.executorService);
+        return new StorageWriterFactory(writerConfig, this.coreExecutor);
```

Comment:
Should this be `storageExecutor`?

Reply:
No. StorageWriter is part of the Core SegmentStore. It is not a Storage abstraction (i.e. it does not implement `Storage`).